tests/omamqp1-basic.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+# This file is part of the rsyslog project, released under ASL 2.0
+. ${srcdir:=.}/diag.sh init
+. $srcdir/omamqp1-common.sh
+
+AMQP_URL=${AMQP_URL:-"localhost:5672"}
+NUMMESSAGES=10240
+
+generate_conf
+if [ "${USE_VALGRIND:-}" = YES ] ; then
+	add_conf '
+global(debug.unloadModules="off")
+'
+fi
+add_conf '
+template(name="tpl" type="string"
+	 string="{\"msgnum\":\"%msg:F,58:2%\"}")
+
+module(load="../plugins/impstats/.libs/impstats" interval="1"
+	   log.file="'"$RSYSLOG_DYNNAME.spool"'/stats.log" log.syslog="off" format="cee")
+module(load="../contrib/omamqp1/.libs/omamqp1")
+
+if $msg contains "msgnum:" then
+	action(type="omamqp1"
+	       host="'"$AMQP_URL"'"
+	       target="amq.rsyslogtest")
+'
+qdrouterd > $RSYSLOG_DYNNAME.spool/qdrouterd.log 2>&1 & qdrouterdpid=$!
+sleep 5 # give qdrouterd a chance to start up and listen
+# have to start reader before writer - like a pipe - for the client, NUMMESSAGES could
+# be much less than the number of records if batching is used - for the client, it means
+# batches, not records - but there can't be more batches than records
+amqp_simple_recv $AMQP_URL amq.rsyslogtest $NUMMESSAGES > $RSYSLOG_DYNNAME.spool/amqp_simple_recv.out 2>&1 &
+#export RSYSLOG_DEBUG=debug
+#export RSYSLOG_DEBUGLOG=/tmp/rsyslog.debug.log
+startup
+if [ "${USE_VALGRIND:-}" = YES ] ; then
```

Comment:
This should not be necessary, as the testbench framework handles this. Do you have seen some cases where a problem occurred? If so, I'd like to fix that in the plumbing.

Reply:
> Yes, the problem is that you do not get a full valgrind stack trace with the full symbols for leak detection because the module is unloaded before valgrind can report the leaks.

actually, that was not the point of my question ;-) That `if` guards a `sleep 5` which I think is not necessary. But I'll also check how the rest of the testbench handles module unload. I think it is automatically handled in valgrind tests.

Reply:
ah, I see - yes, it seems that rsyslog needed a little longer to start up when using valgrind, otherwise, the test would not pass

Reply:
OK, good to know. Looks like a problem with the testbench plumbing. I'll merge the PR with that sleep and then will try to reproduce the issue with a separate PR.